### ISSUE TITLE
NavigationViewService: GetSelectedItem is looking on MenuItems, but not on MenuItemsSource

### ISSUE THREAD

**Issue by tsvietOK** [microsoft/TemplateStudio#4616]
### Describe the bug

When you use binding in NavigationView with property MenuItemsSource, GetSelectedItem is looking for this selected item only in _navigationView.MenuItems and  _navigationView.FooterMenuItems. But in case of binding usage, these two properties will be empty.
![image](https://user-images.githubusercontent.com/20501502/211099067-3b7553b8-8807-499f-971b-7b5e3f441f2c.png)
As a result we have broken navigation selector, but navigation still works. Of course I've tried to change

```cs
return GetSelectedItem(_navigationView.MenuItems, pageType)
```

to

```cs
return GetSelectedItem((IEnumerable<object>)_navigationView.MenuItemsSource, pageType)
```

But in the moment, when app starts, MenuItemSource is null.
![image](https://user-images.githubusercontent.com/20501502/211099773-a225a30d-b24f-458c-8d47-b2b7cd85e37f.png)

### To Reproduce

I've created a test [app](https://github.com/tsvietOK/NavigationTestApp). App in `master` branch is just output form TemplateStudio + added MenuItemsSource to NavView. App in `broken` branch is the same as master branch, but with changed `GetSelectedItem(_navigationView.MenuItems, pageType)` to `GetSelectedItem((IEnumerable<object>)_navigationView.MenuItemsSource, pageType)`.

### Applies to the following platforms:

- [X] WinUI
- [ ] WPF
- [ ] UWP

### About your setup

 - Visual Studio Version:  17.4.3
 - Template Studio Wizard Version: 5.2.1
 - Windows Build: 22621.963

**Comment by mikebattista** [comment]
Is this a timing issue? Is MenuItems eventually populated when you use MenuItemsSource?

Also you're trying to cast a MenuItemsSource to an IEnumerable to get the list of MenuItems. Is that documented behavior or just something you tried? I wouldn't expect that to work.

**Comment by tsvietOK** [comment]
> Is this a timing issue? Is MenuItems eventually populated when you use MenuItemsSource?
Yeah, I think so. MenuItemsSource is not available only at the moment when app starts. I think this this the reason why there is no selected item after startup. As temporary fix I've added `if (_navigationView != null && _navigationView.MenuItemsSource != null)` MenuItemsSource null check.
> 
> Also you're trying to cast a MenuItemsSource to an IEnumerable to get the list of MenuItems. Is that documented behavior or just something you tried? I wouldn't expect that to work.
Yes, this is because `GetSelectedItem` is expecting to get IEnumerable<>, but MenuItemsSource is an object, not collection.

**Comment by mikebattista** [comment]
GetSelectedItem is expecting a collection, yes, but MenuItemsSource isn't a collection as you said. It's an object.

Are you saying that you've confirmed that if MenuItemsSource isn't null, casting it to an IEnumerable will return a collection of MenuItems?

**Comment by tsvietOK** [comment]
No, I am saying that when you use MenuItems defined in XAML, then it (navigation view) will use _navigationView.MenuItems property (MenuItemsSource is null in this case) and navigation works fine. If you use MenuItemsSource in XAML, it (navigation view) will use _navigationView.MenuItemsSource property and _navigationView.MenuItems will be empty. But method GetSelectedItem still looking for selected item in MenuItems.

**Comment by tsvietOK** [comment]
Of course in order to fix that when moved from MenuItems to MenuItemsSource in method GetSelectedItem I've added null check for MenuItemsSource and added cast to IEnumerable<> and now it is working (there is still no selected item on startup).

**Comment by mikebattista** [comment]
I understand the problem you reported. I don't expect the workaround you tried (to cast MenuItemsSource to an IEnumerable) is a valid workaround.

If it's just a timing issue, there are a couple tactics to try:

1. Use [DispatcherQueue](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.dispatching.dispatcherqueue.tryenqueue?view=windows-app-sdk-1.2#microsoft-ui-dispatching-dispatcherqueue-tryenqueue(microsoft-ui-dispatching-dispatcherqueuepriority-microsoft-ui-dispatching-dispatcherqueuehandler)) with Low priority to queue the operation. MenuItem properties may be populated when the queued callback executes. The sample App Notification code in the templates demonstrates this.
2. Wire up a Page.OnLoaded event and update the selected item then. The ShellPage.xaml.cs in the templates already wires up this event to update the TitleBar and KeyboardAccelerators for example so you could add some code there.
